tests/statistics/stats_utils.py

```diff
@@ -0,0 +1,105 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from collections import OrderedDict
+from datetime import datetime
+
+import pytest
+
+from pytest_pootle.factories import UserFactory
+
+from django.contrib.auth import get_user_model
+from django.db.models import Q
+
+from pootle.core.delegate import contributors
+from pootle_statistics.models import Submission, SubmissionTypes
+from pootle_statistics.utils import Contributors
+from pootle_store.models import Store
+
+
+User = get_user_model()
+
+
+def _contributors_list(contribs,
+                       projects=None, languages=None,
+                       since=None, until=None):
+    subs = []
+    for user in contribs.user_qs.distinct():
+        q = Q(submitter=user)
+        if projects:
+            q = q & Q(
+                translation_project__project__code__in=projects)
+        if languages:
+            q = q & Q(
+                translation_project__language__code__in=languages)
+        if since or until:
+            _q = Q()
+            if since:
+                _q = _q & Q(creation_time__gte=since)
+            if until:
+                _q = _q & Q(creation_time__lte=until)
+            q = q & _q
+        submissions = Submission.objects.filter(q)
+        if submissions.count():
+            subs.append((user.username, submissions.count()))
+    return OrderedDict(sorted(subs, key=lambda i: (-i[1], i[0])))
```

Comment:
We do not necessarily want to sort like this. If sorting by name it would be only `key=lambda i: i[0]`.